src/jdk.jfr/share/classes/jdk/jfr/internal/EventInstrumentation.java

```diff
@@ -222,6 +246,30 @@ private void methodShouldCommit(CodeBuilder codeBuilder) {
             codeBuilder.invokevirtual(eventClassDesc, sd.methodName(), mdesc);
             codeBuilder.ifeq(fail);
         }
+        if (throttled) {
+            // long d =  eventConfiguration.throttle(this.duration);
+            // this.duration = d;
+            // if (d & MASK_THROTTLE_BIT == 0) {
+            //   goto fail;
+            // }
+            getEventConfiguration(codeBuilder);
+            codeBuilder.aload(0);
+            getfield(codeBuilder, eventClassDesc, ImplicitFields.FIELD_START_TIME);
+            codeBuilder.aload(0);
+            getfield(codeBuilder, eventClassDesc, ImplicitFields.FIELD_DURATION);
+            Bytecode.invokevirtual(codeBuilder, TYPE_EVENT_CONFIGURATION, METHOD_THROTTLE);
+            int result = codeBuilder.allocateLocal(TypeKind.LONG);
```

Comment:
Do we really need to store the result in a local? Can't we just dup it on the expression stack and store it directly into the field after another aload, or dup? Perhaps dup twice to then issue the mask operation?

Reply:
No big deal, we can look into this later if the need arises.